bin/plugin/commands/packages.js

```diff
@@ -431,13 +445,26 @@ async function publishPackagesToNpm( {
 		);
 
 		log( '>> Publishing modified packages to npm.' );
-		await command(
-			`npx lerna publish from-package ${ yesFlag } ${ noVerifyAccessFlag }`,
-			{
-				cwd: gitWorkingDirectoryPath,
-				stdio: 'inherit',
-			}
-		);
+		try {
+			await command(
+				`npx lerna publish from-package ${ yesFlag } ${ noVerifyAccessFlag }`,
+				{
+					cwd: gitWorkingDirectoryPath,
+					stdio: 'inherit',
+				}
+			);
+		} catch {
+			log(
+				'>> Trying to finish failed publishing of modified npm packages.'
+			);
+			await command(
```

Comment:
We will also need to run `git reset --hard` before trying `npx lerna publish from-package`. I ran into this issue today:

<img width="993" alt="Screenshot 2023-08-17 at 11 42 50" src="https://github.com/WordPress/gutenberg/assets/699132/560cf757-e872-4d40-8efb-e4511cc833e8">

Reply:
Changes added in https://github.com/WordPress/gutenberg/pull/53565/commits/a6356c84e937ca5516b1af0182e293436c142cb0.